src/Core/Checkout/Customer/CustomerException.php

```diff
@@ -283,4 +291,65 @@ public static function invalidImitationToken(string $token): InvalidImitateCusto
     {
         return new InvalidImitateCustomerTokenException($token);
     }
+
+    public static function missingRouteAnnotation(string $annotation, string $route): self
+    {
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            self::MISSING_ROUTE_ANNOTATION,
+            'Missing @{{ annotation }} annotation for route: {{ route }}',
+            ['annotation' => $annotation, 'route' => $route]
+        );
+    }
+
+    public static function missingRouteSalesChannel(string $route): self
+    {
+        return new self(
+            Response::HTTP_INTERNAL_SERVER_ERROR,
+            self::MISSING_ROUTE_SALES_CHANNEL,
+            'Missing sales channel context for route {{ route }}',
+            ['route' => $route]
+        );
+    }
+
+    public static function unsupportedValue(string $type, string $class): self
+    {
+        return new self(
+            Response::HTTP_BAD_REQUEST,
+            self::VALUE_NOT_SUPPORTED,
+            'Unsupported value of type {{ type }} in {{ class }}',
+            ['type' => $type, 'class' => $class]
+        );
+    }
+
+    public static function missingRequestParameter(string $name): self
+    {
+        return new self(
+            Response::HTTP_BAD_REQUEST,
+            self::MISSING_REQUEST_PARAMETER_CODE,
+            'Parameter "{{ parameterName }}" is missing.',
+            ['parameterName' => $name]
+        );
+    }
+
+    public static function productNotFound(string $productId): ShopwareHttpException
+    {
+        return new ProductNotFoundException($productId);
```

Comment:
we should throw a specific exception from another domain

Reply:
I changed it to throw CustomerException. When throwing an exception from another domain it breaks the Domain Exception 'pattern'. And PHPStan complains.